crates/sparrow-merge/src/gather.rs

```diff
@@ -77,6 +81,11 @@ impl Gatherer {
         self.active.is_empty() || self.active.peek().unwrap().up_to_time > self.emitted_up_to_time
     }
 
+    /// Returns true if all inputs have been closed.
+    pub fn all_closed(&self) -> bool {
+        self.pending.iter().all(|p| p.closed)
```

Comment:
This makes sense for now, but maybe worth a quick comment that if this ever shows up as taking time (such as if there are lots of pending items), it may be better to count the number of "pending" items separately, so this can just be `self.pending_count == 0` or something.